src/main/groovy/com/brightsparklabs/gradle/baseline/BaselinePlugin.groovy

```diff
@@ -351,45 +358,139 @@ public class BaselinePlugin implements Plugin<Project> {
 
         // Error early if configuration in invalid.
         if (bucketNameIsEmpty) {
-            def error = "`bslGradle.deploy.s3.bucketName` cannot be null or empty. Value was: `${bucketName}`"
+            def error = "`bslGradle.deploy.s3.bucketName` cannot be null or empty. Value was:" +
+                    " `${bucketName}`"
             project.logger.error(error)
             throw new IllegalStateException(error)
         }
         if (filesToUploadIsEmpty) {
-            def error = "`bslGradle.deploy.s3.filesToUpload` cannot be null or empty. Value was: ${filesToUpload}"
+            def error = "`bslGradle.deploy.s3.filesToUpload` cannot be null or empty. Value was:" +
+                    " ${filesToUpload}"
             project.logger.error(error)
             throw new IllegalStateException(error)
         }
 
+        // Get the absolute paths of the files to upload. We search for files in the `build`
+        // directory, and pattern match against their absolute paths.
+        final Set<String> filesToUploadPaths = []
+        for (String fileRegex in filesToUpload) {
+            project.buildDir.eachFileRecurse(FileType.FILES) { File file ->
```

Comment:
This will crawl the filesystem for each regex. Which will be slow due to disk IO.

Build a list of all files first, then iterate through that list:
```
def allBuildFiles = [] 
project.buildDir.eachFileRecurse(FileType.FILES) { File file -> allBuildFiles.add(file) }

for (String fileRegex in filesToUpload) {
  allBuildFiles.each (...)
}
```